tests/mpmc_channel.rs

```diff
@@ -368,6 +387,38 @@ macro_rules! gen_mpmc_tests {
                 assert!(fut2.is_terminated());
             }
 
+            #[test]
+            fn stream_smoke_test() {
+                let channel = ChannelType::new();
+                let (waker, count) = new_count_waker();
+                let cx = &mut Context::from_waker(&waker);
+
+                let stream = channel.stream();
+                pin_mut!(stream);
+
+                let fut = stream.next();
+                pin_mut!(fut);
+                assert!(fut.as_mut().poll(cx).is_pending());
+                assert_eq!(count, 0);
+
+                assert_send(cx, &channel, 1);
+
+                assert_receive_stream_done(cx, &mut fut, Some(1));
+                assert_eq!(count, 1);
+
+                let fut = stream.next();
+                pin_mut!(fut);
+                assert!(fut.as_mut().poll(cx).is_pending());
+                assert_eq!(count, 1);
+
+                channel.close();
+
+                assert_receive_stream_done(cx, &mut fut, None);
```

Comment:
Maybe add another one of these. I think any following receive call should also return `None`

Reply:
Done. I should probably add more meaningful tests however.